lib/Target/ExportSystemC/Patterns/EmitCEmissionPatterns.cpp

```diff
@@ -0,0 +1,43 @@
+//===- EmitCEmissionPatterns.cpp - EmitC Dialect Emission Patterns --------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// This implements the emission patterns for the emitc dialect.
+//
+//===----------------------------------------------------------------------===//
+
+#include "EmitCEmissionPatterns.h"
+#include "../EmissionPrinter.h"
+#include "mlir/Dialect/EmitC/IR/EmitC.h"
+
+using namespace mlir::emitc;
+using namespace circt;
+using namespace circt::ExportSystemC;
+
+//===----------------------------------------------------------------------===//
+// Operation emission patterns.
+//===----------------------------------------------------------------------===//
+
+namespace {
+/// Emit emitc.include operations.
+struct IncludeEmitter : OpEmissionPattern<IncludeOp> {
+  using OpEmissionPattern::OpEmissionPattern;
+  void emitStatement(IncludeOp op, EmissionPrinter &p) override {
+    p << "#include " << (op.getIsStandardInclude() ? "<" : "\"")
+      << op.getInclude() << (op.getIsStandardInclude() ? ">" : "\"") << "\n";
+  }
+};
```

Comment:
I really think this is an interesting thing to bring up on MLIR discourse. You're needing to define a custom emitter for an op _in another dialect_. It would be cool to see this packaged up with EmitC such that its definition of emission of includes could be re-used.

This is, admittedly, entirely over-engineered and not having an immediate use case. (It is very rare to need to access to another dialect's printing.)